tcp/mux.go

```diff
@@ -203,8 +207,20 @@ func (ln *listener) Accept() (c net.Conn, err error) {
 	return conn, nil
 }
 
-// Close is a no-op. The mux's listener should be closed instead.
-func (ln *listener) Close() error { return nil }
+// Close removes this listener from the parent mux and closes the channel.
+func (ln *listener) Close() error {
+	ln.mux.mu.Lock()
```

Comment:
Personally I don't like objects reaching into their Fields' muxtexes and controlling them. I'd rather see a `Close() error` method defined on `*Mux` that takes listener and iterates over it's map of listeners, looking for the right one to close.

Reply:
To be fair, this listener is not exposed and other places in the listener reach into the mux's mutex. I'll try to rearrange it though.